fcrepo-http-api/src/main/java/org/fcrepo/http/api/FedoraLdp.java

```diff
@@ -434,7 +434,12 @@ public Response updateSparql(@ContentLocation final InputStream requestBodyStrea
             }
             session.save();
 
-            addCacheControlHeaders(servletResponse, resource(), session);
+            if (resource() instanceof FedoraBinary) {
```

Comment:
Yes, because that logic is specific to handling PATCH operations on NonRdfSourceDescriptions -- for PUT and POST, NonRdfSourceDescriptions don't apply.

N.B. I realize that this code is checking for an `instanceof` a FedoraBinary. I actually do not understand why that works, but it is actually quite necessary. You will notice in the code above (in the same function), there is an earlier check for FedoraBinary, which returns false. Somehow, after calling `session.save()` the FedoraResource discovers that it is a FedoraBinary and not a NonRdfSourceDescription.